src/utils/configured-graphql-client.ts

```diff
@@ -0,0 +1,17 @@
+import {GraphQLClient} from 'graphql-request'
+import config from '../lib/config'
+import getAccessTokenFromCookie from './get-access-token-from-cookie'
+
+export const getGraphQLClient = () => {
+  const graphQLClient = new GraphQLClient(config.graphQLEndpoint, {
+    headers: config.headers,
+  })
+
+  const authToken = getAccessTokenFromCookie()
+
+  if (authToken) {
+    graphQLClient.setHeader('Authorization', `Bearer ${authToken}`)
+  }
+
+  return graphQLClient
```

Comment:
consolidated the graphql client to this function. It will automatically add the `x-site-client` header and add the auth token as well if its available.